plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoIpProcessor.java

```diff
@@ -209,4 +262,22 @@ public AddressNotFoundRuntimeException(Throwable cause) {
         }
     }
 
+    public enum Field {
+
+        IP,
+        COUNTRY_ISO_CODE,
+        COUNTRY_NAME,
+        CONTINENT_NAME,
+        REGION_NAME,
+        CITY_NAME,
+        TIMEZONE,
+        LATITUDE,
+        LONGITUDE,
+        LOCATION;
+
+        public static Field parse(String value) {
+            return valueOf(value.toUpperCase(Locale.ROOT));
```

Comment:
do we want to catch the potential `IllegalArgumentException` or `NPE`, or leave the default exception message that will be thrown?